Firefox/webextension/background.js

```diff
@@ -4,18 +4,17 @@
 
 // Ask to the legacy part to dump the needed data and send it back
 // to the background page...
-var port = browser.runtime.connect({name: "sync-legacy-addon-data"});
-port.onMessage.addListener((msg) => {
-	if (msg) {
-		browser.storage.local.get('isMigrated').then(function(item) {
-			if (item.isMigrated) {
-				console.warn('Skipping migration');
-				return;
-			}
+browser.storage.local.get('isProperlyMigrated').then(function(item) {
+	if (item.isProperlyMigrated) {
+		console.warn('Skipping migration', item);
+		return;
+	}
+
+	var port = browser.runtime.connect({name: "sync-legacy-addon-data"});
+	port.onMessage.addListener((msg) => {
+		if (msg) {
 			// The corresponding code in XKit is GM_setValue(name, value) -> set({name: value})
-			msg.isMigrated = true;
-			console.log('Migrating pref storage', msg);
 			browser.storage.local.set(msg);
```

Comment:
Can we make sure we're:

a) not getting any errors from the `.set` call—I think people were getting "invalid value" errors and some were definitely getting quota issues
b) not over-writing non-empty values? I'm not sure exactly what the best value of "non-empty" here is, and this may be too much work, but if this takes us a day or two to get out, then that's potentially a day of missing data for the people the migration worked for.